### PR TITLE
Minor fixes and cosmetic changes

### PR DESCRIPTION
**Pull request by dev-experience**
Updates:
- Implement generic `.As<T>()` in addition to `.As(Type type)`. And fix nullable warnings

Minor fixes:
- Fix transitive dependency warning for `Microsoft.AspNetCore.WebUtilities` v2.2.0: "Transitive dependency System.Text.Encodings.Web 4.5.0 contains vulnerabilities according to Checkmarx". Explicitly install `System.Text.Encodings.Web` 4.7.2
- Fix potential `NullReferenceException`. Throw `RequestCreationException` instead
- Fix warning `CS8604 - Possible null reference argument for parameter`
- Fix broken XML comment
- Fix nullability warnings
- Minor improvements in error messages
- Fix multiple enumeration warnings for `IEnumerable<T>`

Various cosmetic changes:

- Fix casing to camel case in URL params
- Minor changes in XML docs texts
- Fix formatting of long lines
- Fix warning for casing in a test name
- Remove redundant property initialization
- Minor cosmetic changes in the code
- Fix tabs in `csproj` file

### CODE REVIEW

**Comment by basdijkstra** [comment]
Hey @dev-experience, thank you so much for this PR, and the other ones, too. I really appreciate them!

I’ll have a good look at all of them soon, but please give me a couple of days.

**Comment by basdijkstra** [comment]
@dev-experience, I've merged the first two PRs you submitted, I think this one needs to be rebased first and have the merge conflicts solved. Could you please look into that?

**Comment by dev-experience** [comment]
> I’ll have a good look at all of them soon, but please give me a couple of days.

@basdijkstra, sure, take your time.

> merge conflicts solved. Could you please look into that?
Sure thing. 
I synced changes from your `main` to my fork, merged changes to the `fix/cosmetic-changes` branch, fixed conflicts.

But I see that checks are still failing. Trying to figure out why.

**Comment by dev-experience** [comment]
OK, I see. Some tests are failing now.
I checked them before making the pull request. Looks like it's after resolving conflicts. I'll fix them this week.

**Comment by basdijkstra** [comment]
Thanks!

**Comment by dev-experience** [comment]
@basdijkstra, clearly, I haven't run tests properly. As there are obvious mismatches in error messages after my changes. Not sure how it happened. Sorry about that.

Tests are fixed now.

**Comment by dev-experience** [comment]
@basdijkstra, still, please review my changes and give any feedback. Some of them can be questioned and I'm open to a discussion. I can revert questionable ones to get accepted ones merged as soon as possible. Please let me know.

**Comment by basdijkstra** [comment]
@dev-experience thanks for fixing! I’ll have a look ASAP, probably in the next day or two.

**Comment by basdijkstra** [comment]
@dev-experience can you remove this commit: https://github.com/basdijkstra/rest-assured-net/pull/52/commits/5ec6426054da08d3f9a7a100c5f8154f40631863 from this PR and submit that as a separate one? I think it's a good addition but it doesn't really fit in with the others as it's much more than a cosmetic change.

Also:
* Why did you explicitly set the language version to 10? That discards .NET Core 3.1, which at the moment is still supported.
* Could you please also provide an acceptance test that demonstrates the use of `As<T>()`? Preferably in https://github.com/basdijkstra/rest-assured-net/blob/main/RestAssured.Net.Tests/ResponseBodyDeserializationTests.cs

Will look at the others right now.

File: RestAssured.Net/Response/VerifiableResponse.cs [hunk 1]
```diff
@@ -441,7 +441,7 @@ public object As(Type type)
 
             if (responseMediaType == null || responseMediaType.Contains("json"))
             {
-                return JsonConvert.DeserializeObject(this.response.Content.ReadAsStringAsync().Result, type) ?? string.Empty;
```
Comment (basdijkstra) [inline RestAssured.Net/Response/VerifiableResponse.cs:450]: Removing the `string.Empty` generates a 'possible null reference return` here, that's why it's there.

File: RestAssured.Net/Response/VerifiableResponse.cs [hunk 2]
```diff
@@ -87,7 +87,7 @@ public VerifiableResponse StatusCode(int expectedStatusCode)
         {
             if (expectedStatusCode != (int)this.response.StatusCode)
             {
-                throw new ResponseVerificationException($"Expected status code to be {expectedStatusCode}, but was {(int)this.response.StatusCode}");
+                throw new ResponseVerificationException($"Expected status code to be '{expectedStatusCode}', but was '{(int)this.response.StatusCode}'");
```
Comment (basdijkstra) [inline RestAssured.Net/Response/VerifiableResponse.cs:90]: These are integers, not strings, so I don't think quotes are needed here. I prefer them without. Applies to all occurrences.

**Comment by basdijkstra** [comment]
Hey @dev-experience, I like a lot of the changes you provided, but not all of them. Instead of going back and forth on all of them, how about I make the changes you provided myself and commit them? There's a lot of commits in this one and I think this would be much more effective.

I'll obviously credit you for all the suggestions :)

Oh and please still submit that separate PR regarding commit https://github.com/basdijkstra/rest-assured-net/commit/5ec6426054da08d3f9a7a100c5f8154f40631863 I mentioned earlier, I'd love that.

**Pull request closed without merging**